### PR TITLE
Block log update

### PR DESCRIPTION
**Pull request by hkalodner**
- Update ArbOS with https://github.com/OffchainLabs/arb-os/pull/246
- Add support for updated block log format

### CODE REVIEW

File: packages/arb-evm/evm/blockresult.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	ChainStats    *OutputStatistics
+	PreviousBlock *big.Int
+}
+
+func (b *BlockInfo) LastAVMLog() *big.Int {
+	return new(big.Int).Sub(b.ChainStats.AVMLogCount, big.NewInt(1))
+}
+
+func (b *BlockInfo) FirstAVMLog() *big.Int {
+	return new(big.Int).Sub(b.LastAVMLog(), b.BlockStats.AVMLogCount)
```
Comment (edfelten) [inline packages/arb-evm/evm/blockresult.go:59]: It looks like you need to add 1 to this.  If the logs in this block are numbers 10, 11, and 12, then LastAVMLog() will be 12, AVMLogCount will be 3, so FirstAVMLog will return 9.  It should return 10.

File: packages/arb-evm/evm/blockresult.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+func (b *BlockInfo) LastAVMSend() *big.Int {
+	return new(big.Int).Sub(b.ChainStats.AVMSendCount, big.NewInt(1))
+}
+
+func (b *BlockInfo) FirstAVMSend() *big.Int {
+	return new(big.Int).Sub(b.LastAVMSend(), b.BlockStats.AVMSendCount)
```
Comment (edfelten) [inline packages/arb-evm/evm/blockresult.go:67]: Same off-by-one bug as above -- should add 1 to this.

**Review by edfelten — CHANGES_REQUESTED** [review]
Found a couple of apparent bugs.

**Comment by hkalodner** [comment]
I'm splitting this into two PRs so I'm closing this one

Backwards compatible changes are in https://github.com/OffchainLabs/arbitrum/pull/668
ArbOS breaking changes will be in a new PR

**Pull request closed without merging**